File: frontend/src/domain/validations/auth/EmailValidation.ts
import type { IValidationStrategy } from "@validations/IValidationStrategy";
import { REGEX } from "@constants/regex";
import { DATA_TYPES } from "@constants/dataTypes";

export class EmailValidation implements IValidationStrategy {
  validate(input: unknown): boolean {
    if (typeof input !== DATA_TYPES.STRING) {
      return false;
    }
    return REGEX.EMAIL.test(input as string);
  }
}

File: backend/src/domain/validations/email_validation.py
import re
from domain.validations.i_validation_strategy import IValidationStrategy


class EmailValidation(IValidationStrategy):
    """Email validation using regex pattern."""

    EMAIL_REGEX = re.compile(r"^[a-zA-Z0-9._%+-]+@[a-zA-Z0-9.-]+\.[a-zA-Z]{2,}$")

    def validate(self, input_value: any) -> bool:
        """Validate email format."""
        if not isinstance(input_value, str):
            return False
        return bool(self.EMAIL_REGEX.match(input_value))
